test/test.go

```diff
@@ -102,61 +117,88 @@ func findImages(testDir string) (imageFilePaths []string) {
 // ParseTestResults ...
 func ParseTestResults(testsRootDir string) (results Results, err error) {
 	// read dirs in base tests dir
+	// <root_tests_dir>
+
 	testDirs, err := ioutil.ReadDir(testsRootDir)
 	if err != nil {
 		return nil, err
 	}
 
 	// find test results in each dir, skip if invalid test dir
 	for _, testDir := range testDirs {
-		testDirPath := filepath.Join(testsRootDir, testDir.Name())
+		if !testDir.IsDir() {
+			continue
+		}
 
-		// read one level of file set only <root_tests_dir>/<test_dir>/file_to_get
-		testFiles, err := filepath.Glob(filepath.Join(testDirPath, "*"))
+		// <root_tests_dir>/<test_dir>
+		testDirPath := filepath.Join(testsRootDir, testDir.Name())
+		// read unique test phase dirs
+		testPhaseDirs, err := ioutil.ReadDir(testDirPath)
 		if err != nil {
 			return nil, err
 		}
 
-		// find step-info in dir
+		// find step-info in dir, continue if no step-info.json as this file is only required if step has exported artifacts also
+		// <root_tests_dir>/<test_dir>/step-info.json
 		var stepInfo *models.TestResultStepInfo
-		for _, file := range testFiles {
-			if strings.HasSuffix(file, "step-info.json") {
-				stepInfoFileContent, err := fileutil.ReadBytesFromFile(file)
-				if err != nil {
-					return nil, err
-				}
-				if err := json.Unmarshal(stepInfoFileContent, &stepInfo); err != nil {
-					return nil, err
-				}
-			}
+		stepInfoFileContent, err := fileutil.ReadBytesFromFile(filepath.Join(testDirPath, "step-info.json"))
+		if err != nil {
+			continue
```

Comment:
i think we should log a warning for debugging reasons here, and maybe above `testDir.IsDir` as well